geopandas/plotting.py

```diff
@@ -445,7 +451,14 @@ def plot_dataframe(df, column=None, cmap=None, color=None, ax=None,
         values = np.array([valuemap[k] for k in values])
 
     if scheme is not None:
-        binning = _mapclassify_choro(values, scheme, k=k)
+        if classification_kwds is None:
+            classification_kwds = {}
+        if scheme not in ['box_plot', 'headtail_breaks', 'percentiles',
+                          'std_mean', 'user_defined']:
+            if 'k' not in classification_kwds:
+                classification_kwds['k'] = k
+
```

Comment:
@jdmcbr the reason for explicit list was this. Some of the classifiers cannot take k as a number of bins, but I will try to handle exception instead and do it according to your comments.